### PR TITLE
Remove tenant id from header in endpoint resolution request.

### PR DESCRIPTION
**Pull request by tarostok**

### CODE REVIEW

File: Backend/src/Services/AuthenticationService.cs [hunk 1]
```diff
@@ -156,19 +171,26 @@ public static void AddBearerClaimToResponse(MsalUiRequiredException ex, HttpResp
...
+                else
+                {
+                    return new AuthorizationContext { };
+                }
             }
 
             var subjectClaims = ValidateAadTokenCommon(subjectAndAppToken.SubjectToken, openIdConnectConfiguration, isAppOnly: false);
             ValidateClaimValue(subjectClaims, "appid", appTokenAppId, "subject and app tokens should belong to same application");
-            ValidateClaimValue(subjectClaims, "tid", subjectTenantId.ToString(), "subject tokens must belong to the subject's tenant");
+            ValidateClaimValue(subjectClaims, "tid", subjectTenantId.Value.ToString(), "subject tokens must belong to the subject's tenant");
```
Comment (atefsaw) [inline Backend/src/Services/AuthenticationService.cs:186]: Shouldn't we validate if subjectTenantId has value first? (In the case of subjectTenantId is null and requireTenantIdHeader is false)

**Review by atefsaw — APPROVED** [review]

**Comment by atefsaw** [comment]
@tarostok , what's the motivation of removing the tenant-id header in the ER endpoint?

File: Backend/src/Services/AuthenticationService.cs (hunk 1 above)
Comment (tarostok) [reply Backend/src/Services/AuthenticationService.cs:186]: There is a check at line 154 - if (requireTenantIdHeader && !subjectTenantId.HasValue)

**Comment by tarostok** [comment]
> @tarostok , what's the motivation of removing the tenant-id header in the ER endpoint?



> @tarostok , what's the motivation of removing the tenant-id header in the ER endpoint?

For endpoint resolution we would like to pass the tenant id at the request body to allow it to be cached and we don't want the workload to resolve the endpoint using the tenant id in the header which can cause cache misalignment.

**Pull request merged**